### PR TITLE
Address Bug in Registry access Read/Write operations

### PR DESCRIPTION
**Pull request by hulumane**
This fixes an issue where we assumed that Registry operations return HRESULTS rather than Win32 Status error messages. We have identified and fixed this issue.

Unit tests pass after fixing some bugs.

### CODE REVIEW

File: dev/AppNotifications/AppNotificationUtility.cpp [hunk 1]
```diff
@@ -274,11 +279,11 @@ void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring
...
 {
     std::wstring registeredGuid;
-    HRESULT status = GetActivatorGuid(registeredGuid);
+    auto hr = GetActivatorGuid(registeredGuid);
 
-    THROW_HR_IF(status, FAILED(status) && status != HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND));
+    THROW_HR_IF(hr, FAILED(hr) && hr != HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND));
 
-    if (status == ERROR_FILE_NOT_FOUND)
+    if (hr == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND))
```
Comment (hulumane) [inline dev/AppNotifications/AppNotificationUtility.cpp:286]: This can be optimized.

File: dev/AppNotifications/AppNotificationUtility.h [hunk 2]
```diff
@@ -32,7 +32,7 @@ namespace Microsoft::Windows::AppNotifications::Helpers
 
     inline void RegisterValue(wil::unique_hkey const& hKey, PCWSTR const& key, const BYTE* value, DWORD const& valueType, size_t const& size)
     {
-        THROW_IF_FAILED(RegSetValueExW(hKey.get(), key, 0, valueType, value, (DWORD)size));
+        THROW_IF_WIN32_ERROR(RegSetValueExW(hKey.get(), key, 0, valueType, value, (DWORD)size));
```
Comment (danielayala94) [inline dev/AppNotifications/AppNotificationUtility.h:35]: nit: `static_cast<DWORD>`

**Review by danielayala94 — APPROVED** [review]

**Comment by loneursid** [comment]
I'm thinking, should we still bother putting stuff into the L1 or should we just send it directly into the PR branch? The PR branch is here: user/erlangl/RI_WNP_Notifications_L1. The original L1 is really of no use at this point (other than from a history perspective.)

**Comment by hulumane** [comment]
> I'm thinking, should we still bother putting stuff into the L1 or should we just send it directly into the PR branch? The PR branch is here: user/erlangl/RI_WNP_Notifications_L1. The original L1 is really of no use at this point (other than from a history perspective.)

@loneursid I'll rebase this if you are fine with that.

File: build/NuSpecs/AppxManifest.xml [hunk 3]
```diff
@@ -55,11 +55,13 @@
...
           <!-- PowerNotifications -->
           <ActivatableClass ActivatableClassId="Microsoft.Windows.System.Power.PowerManager" ThreadingModel="both" />
-		  
-		  <!-- ToastNotifications -->
-		  <ActivatableClass ActivatableClassId="Microsoft.Windows.ToastNotifications.ToastActivationInfo" ThreadingModel="both" />
-		  <ActivatableClass ActivatableClassId="Microsoft.Windows.ToastNotifications.ToastAssets" ThreadingModel="both" />
-		  <ActivatableClass ActivatableClassId="Microsoft.Windows.ToastNotifications.ToastNotificationManager" ThreadingModel="both" />
+
+          <!-- AppNotifications -->
+          <ActivatableClass ActivatableClassId="Microsoft.Windows.AppNotifications.AppNotificationActivationInfo" ThreadingModel="both" />
```
Comment (hulumane) [inline build/NuSpecs/AppxManifest.xml:60]: Delete

**Review by pmpurifoy — APPROVED** [review]
Thanks for fixing this :)

**Pull request merged**